Mu2eG4/src/Mu2eSpecialCutsProcess.cc

```diff
@@ -0,0 +1,165 @@
+                                   const G4Track& ,
+                                   const G4Step&
+                                   ) { return nullptr; };
+
+  // PostStep (not AtRest)
+
+  G4double Mu2eSpecialCutsProcess::PostStepGetPhysicalInteractionLength(
+                                                                        const G4Track& aTrack,
+                                                                        G4double, // previousStepSize
+                                                                        G4ForceCondition* condition  )
+  {
+    // condition is set to "Not Forced"; if ommited will allways trigger
+    *condition = NotForced;
+
+    G4double proposedStep = std::numeric_limits<double>::max();
+
+    // Maximum number of steps
+
+    if (aTrack.GetCurrentStepNumber() >= static_cast<G4int>(mu2elimits_->maxStepsPerTrack())) {
+      proposedStep = 0.;
+    }
+    if (verboseLevel>0) {
+      G4cout << __func__ << " : "
+             << GetProcessName()
+             << " : current step "
+             << aTrack.GetCurrentStepNumber()
+             << ", max step limit "
+             << static_cast<G4int>(mu2elimits_->maxStepsPerTrack())
+             << ", proposed step "
+             << proposedStep
+             << G4endl;
+    }
+    return proposedStep;
+
+  }
+
+  G4VParticleChange* Mu2eSpecialCutsProcess::PostStepDoIt( const G4Track& aTrack,
+                                                           const G4Step&  )
+  //
+  // Kill the current particle; deposit its kinetic energy based on the cut used
+  //
+  {
+    if (verboseLevel>0) {
+      G4cout << __func__ << " : "
+             << GetProcessName()
+             << " : current step "
+             << aTrack.GetCurrentStepNumber()
+             << ", max step limit "
+             << static_cast<G4int>(mu2elimits_->maxStepsPerTrack())
+             << G4endl;
+    }
+    aParticleChange.Initialize(aTrack);
+    // if the track reached too many steps deposit its energy
+    if (aTrack.GetCurrentStepNumber() >=
+        static_cast<G4int>(mu2elimits_->maxStepsPerTrack())) {
+      aParticleChange.ProposeEnergy(0.) ;
+      aParticleChange.ProposeLocalEnergyDeposit(aTrack.GetKineticEnergy()) ;
+    }
+    // kill the track either way
+    aParticleChange.ProposeTrackStatus(fStopAndKill);
+    return &aParticleChange;
+  }
+
+  // AtRest
+
+  G4double Mu2eSpecialCutsProcess::AtRestGetPhysicalInteractionLength( const G4Track& aTrack,
+                                                                       G4ForceCondition* condition
+                                                                       )
+  {
+    // condition is set to "Not Forced"
+    *condition = NotForced;
+
+    G4double proposedStep = std::numeric_limits<double>::max();
+
+    // Maximum number of steps
+
+    if (aTrack.GetCurrentStepNumber() > static_cast<G4int>(mu2elimits_->maxStepsPerTrack())) {
+      proposedStep = std::numeric_limits<double>::min();
```

Comment:
In AtRestGetPhysicalInteractionLength() we set  proposedStep = std::numeric_limits<double>::min()
while in PostStepGetPhysicalInteractionLength() it was proposedStep = 0.
Why is the difference?

Reply:
decays use double_min, so not to compete with them, but perhaps it should be 2*double_min if we prefer the decays for sure to occur and 0 in cases when we want to prevent them